### ISSUE TITLE
Error when processing x64 binary built with Delphi 11.2.

### ISSUE THREAD

**Issue by andersmelander** [andersmelander/map2pdb#3]
**[Original report](https://bitbucket.org/anders_melander/map2pdb/issues/3) by Hiroyuki Hori (Bitbucket: [Hiroyuki Hori](https://bitbucket.org/Hiroyuki Hori), ).**

----------------------------------------

If you run the attached file run.bat, you will get the following error.

```
F:\Report>run

F:\Report>map2pdb.exe -pdb:Win64\Debug\Project1.pdb -bind:Win64\Debug\Project1.exe -altpdb:"Project1.pdb" Win64\Debug\Project1.map
map2pdb - Copyright (c) 2021 Anders Melander
Version 2.6.2

Exception: Overlapping segments: .data [0002:00000000] and .text [0001:00000000]

F:\Report>
```

‌

**Comment by andersmelander** [comment]
**Original comment by Anders Melander (Bitbucket: [anders_melander](https://bitbucket.org/anders_melander), ).**

----------------------------------------

Reproduced.

The problem is that the segment address values have grown from 8 digit hex values to 16 digit values. The hex parser only reads the first 8 digits so the values all end up as zero.

**Comment by andersmelander** [comment]
**Original comment by Anders Melander (Bitbucket: [anders_melander](https://bitbucket.org/anders_melander), ).**

----------------------------------------

Increased supported width of address offset values from 8 to 16 hex digits in map reader.
Changed address offset type from Cardinal to UInt64.
Fixes #3

**Comment by andersmelander** [comment]
**Original comment by Hiroyuki Hori (Bitbucket: [Hiroyuki Hori](https://bitbucket.org/Hiroyuki Hori), ).**

----------------------------------------

Thank you for the quick fix!
